### PR TITLE
update Ed25519 algorithm number and mnemonic and hook up algorithm 15 in the recursor

### PR DESCRIPTION
**Pull request by mind04**
http://www.iana.org/assignments/dns-sec-alg-numbers/dns-sec-alg-numbers.xhtml
http://www.iana.org/go/draft-ietf-curdle-dnskey-eddsa-03

### Checklist
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code

### CODE REVIEW

File: pdns/dbdnsseckeeper.cc [hunk 1]
```diff
@@ -79,7 +79,7 @@ bool DNSSECKeeper::addKey(const DNSName& name, bool setSEPBit, int algorithm, in
     if(algorithm <= 10)
       throw runtime_error("Creating an algorithm " +std::to_string(algorithm)+" ("+algorithm2name(algorithm)+") key requires the size (in bits) to be passed.");
     else {
-      if(algorithm == 12 || algorithm == 13 || algorithm == 250) // GOST, ECDSAP256SHA256, ED25519SHA512
+      if(algorithm == 12 || algorithm == 13 || algorithm == 15) // GOST, ECDSAP256SHA256, ED25519
```
Comment (Habbie) [inline pdns/dbdnsseckeeper.cc]: Makes me wonder if these should be constants (or references to the enum in dnsseckeeper.hh). Not a task for this PR, of course.

File: regression-tests.api/test_cryptokeys.py [hunk 2]
```diff
@@ -177,7 +177,7 @@ def test_post_wrong_bit_size(self):
         self.assertIn("The algorithm does not support the given bit size.", r.json()['error'])
 
     def test_post_can_not_guess_key_size(self):
-        r = self.add_key(algo=15)
+        r = self.add_key(algo=17)
```
Comment (Habbie) [inline regression-tests.api/test_cryptokeys.py]: should we use a much higher number so we don't go wrong in a year again?

Comment (Habbie) [inline regression-tests.api/test_cryptokeys.py]: ping

Comment (mind04) [inline regression-tests.api/test_cryptokeys.py]: i'm sure the future implementer of algorithm 17 is able to deal with this

**Comment by Habbie** [comment]
good for merge

**Pull request merged**